clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -945,6 +947,8 @@ private void close(KafkaChannel channel, CloseMode closeMode) {
 
         if (idleExpiryManager != null)
             idleExpiryManager.remove(channel.id());
+
+        sensors.maybeUnregisterConnectionMetrics(channel.id());
```

Comment:
I don't agree that we should be removing these metrics when closing a connection.

The per-connection metrics are only created for client-side users of the Selector.

Some of these metrics are meant to be monotonic like the `count` metrics that are instantiated for the Meters. this means we'd be clearing the value any time a client disconnects from a broker.

Given that the number of connection meters instantiated per selector is generally small (scales with # of brokers the selector connects to), I think we should not close these metrics on connection close, and only close when the Selector itself is closed. What do you think?

Reply:
It seems that you are right, David. Closing when the selector is closed makes more sense.